packages/flutter_driver/lib/src/common/create_finder_factory.dart

```diff
@@ -0,0 +1,115 @@
+      case 'PageBack':
+        return _createPageBackFinder();
+      case 'Ancestor':
+        return _createAncestorFinder(finder as Ancestor);
+      case 'Descendant':
+        return _createDescendantFinder(finder as Descendant);
+      default:
+        return null;
+    }
+  }
+
+  Finder _createByTextFinder(ByText arguments) {
+    return find.text(arguments.text);
+  }
+
+  Finder _createByTooltipMessageFinder(ByTooltipMessage arguments) {
+    return find.byElementPredicate((Element element) {
+      final Widget widget = element.widget;
+      if (widget is Tooltip) {
+        return widget.message == arguments.text;
+      }
+      return false;
+    }, description: 'widget with text tooltip "${arguments.text}"');
+  }
+
+  Finder _createBySemanticsLabelFinder(BySemanticsLabel arguments) {
+    return find.byElementPredicate((Element element) {
+      if (element is! RenderObjectElement) {
+        return false;
+      }
+      final String semanticsLabel = element.renderObject?.debugSemantics?.label;
+      if (semanticsLabel == null) {
+        return false;
+      }
+      final Pattern label = arguments.label;
+      return label is RegExp
+          ? label.hasMatch(semanticsLabel)
+          : label == semanticsLabel;
+    }, description: 'widget with semantic label "${arguments.label}"');
+  }
+
+  Finder _createByValueKeyFinder(ByValueKey arguments) {
+    switch (arguments.keyValueType) {
+      case 'int':
+        return find.byKey(ValueKey<int>(arguments.keyValue as int));
+      case 'String':
+        return find.byKey(ValueKey<String>(arguments.keyValue as String));
+      default:
+        throw 'Unsupported ByValueKey type: ${arguments.keyValueType}';
+    }
+  }
+
+  Finder _createByTypeFinder(ByType arguments) {
+    return find.byElementPredicate((Element element) {
+      return element.widget.runtimeType.toString() == arguments.type;
+    }, description: 'widget with runtimeType "${arguments.type}"');
+  }
+
+  Finder _createPageBackFinder() {
+    return find.byElementPredicate((Element element) {
+      final Widget widget = element.widget;
+      if (widget is Tooltip) {
+        return widget.message == 'Back';
+      }
+      if (widget is CupertinoNavigationBarBackButton) {
+        return true;
+      }
+      return false;
+    }, description: 'Material or Cupertino back button');
+  }
+
+  Finder _createAncestorFinder(Ancestor arguments) {
+    final Finder finder = find.ancestor(
+      of: createFinder(arguments.of),
+      matching: createFinder(arguments.matching),
+      matchRoot: arguments.matchRoot,
+    );
+    return arguments.firstMatchOnly ? finder.first : finder;
+  }
+
+  Finder _createDescendantFinder(Descendant arguments) {
+    final Finder finder = find.descendant(
+      of: createFinder(arguments.of),
+      matching: createFinder(arguments.matching),
+      matchRoot: arguments.matchRoot,
+    );
+    return arguments.firstMatchOnly ? finder.first : finder;
+  }
+}
```

Comment:
Please add a new line at the end of the file

Reply:
Applied.